### ISSUE TITLE
PlatformColor crashes app when adding it as dark/light color on Android

### ISSUE THREAD

**Issue by svbutko** [wix/react-native-navigation#7255]
## 🐛 Bug Report

Application crashes on Android when using the new way of adding dark and light theme colors with `PlatformColor`

### Have you read the [Contributing Guidelines on issues](https://wix.github.io/react-native-navigation/docs/meta-contributing/)?

Yes

## To Reproduce

1. Set default options with, for instance `topBar`
2. Add color to `background` like so:
```
        color: {
          light: PlatformColor("?android:attr/colorBackground"),
          dark: PlatformColor("?android:attr/textColorPrimary"),
        },
```

## Expected behavior

Colors are set correctly and app works fine

## Actual Behavior

App crashes:

```
    com.facebook.react.bridge.JSApplicationCausedNativeException: ColorValue: the value must be a number or Object.
        at com.facebook.react.bridge.ColorPropConverter.getColor(ColorPropConverter.java:77)
        at com.reactnativenavigation.options.parsers.ColorParser.parse(ColorParser.kt:33)
        at com.reactnativenavigation.options.params.ThemeColour$Companion.parse(ThemeColour.kt:44)
        at com.reactnativenavigation.options.params.ThemeColour.parse(Unknown Source:2)
        at com.reactnativenavigation.options.TopBarBackgroundOptions.parse(TopBarBackgroundOptions.java:18)
        at com.reactnativenavigation.options.TopBarOptions.parse(TopBarOptions.java:34)
        at com.reactnativenavigation.options.Options.parse(Options.java:22)
        at com.reactnativenavigation.react.NavigationModule.parse(NavigationModule.java:215)
        at com.reactnativenavigation.react.NavigationModule.lambda$setDefaultOptions$1$NavigationModule(NavigationModule.java:120)
        at com.reactnativenavigation.react.-$$Lambda$NavigationModule$vc1HzyGbL5Tqc0F69ZZFUMbcIB0.run(Unknown Source:4)
        at com.reactnativenavigation.react.NavigationModule.lambda$handle$14$NavigationModule(NavigationModule.java:221)
        at com.reactnativenavigation.react.-$$Lambda$NavigationModule$sXUG5B2EQ5skHUFYfoWeiHKR6OE.run(Unknown Source:4)
        at android.os.Handler.handleCallback(Handler.java:938)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loop(Looper.java:223)
        at android.app.ActivityThread.main(ActivityThread.java:7656)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
```

## Your Environment

- React Native Navigation version: 7.20.0
- React Native version: 0.65.1
- Platform(s) (iOS, Android, or both?): Android
- Device info (Simulator/Device? OS version? Debug/Release?): Android Emulator Pixel 5 API 30, Debug

### Are you willing to resolve this issue by submitting a Pull Request?

 - ✅  &nbsp; Yes, I have the time, and I know how to start.
 - ✖️  &nbsp; Yes, I have the time, but I don't know how to start. I would need guidance.
 - ✖️  &nbsp; No, I don’t have the time and I’m okay to wait for the community / maintainers to resolve this issue.

**Comment by svbutko** [comment]
It happens in `ColorParser.kt` when parsing the value is seen as `JSONObject` and when provided to React's `ColorPropConverter.getColor` it's also being send as `JSONObject` but under the hood it's expected to be as instance of `ReadableMap` and because of it there's a crash.

I'm fixing it at this moment

**Comment by swabbass** [comment]
You are using this in the wrong way, look `PlatformColor` is a way to use native colours which also supports dark mode, now
passing PlatformColor to light and dark will cause conflict on how colours work, since when such native colours have night mode and light mode, hence, for each attribute there are 2 possible values, you end up with this:
```
Color :
     light: (native_light1, native_dark1)
     dark: (native_light2, native_dark2)
```
to use PlatformColor, you need to define dark/light natively in XML as Android do, to use our `{dark:.. ,light:...} you can use colors defined in js.

**Comment by svbutko** [comment]
> You are using this in the wrong way, look `PlatformColor` is a way to use native colours which also supports dark mode, now
> passing PlatformColor to light and dark will cause conflict on how colours work, since when such native colours have night mode and light mode, hence, for each attribute there are 2 possible values, you end up with this:
> 
> ```
> Color :
>      light: (native_light1, native_dark1)
>      dark: (native_light2, native_dark2)
> ```
> 
> to use PlatformColor, you need to define dark/light natively in XML as Android do, to use our `{dark:.. ,light:...} you can use colours defined in js.
> 
> We should update the Documentation, to clarify how things working.

Yes but what if I'd like to use already defined values like for example from Material theme without resigning already assigned colors?

`Theme.MaterialComponents.DayNight` already has these colors and for instance I'd like to achieve this:

![image](https://user-images.githubusercontent.com/14828004/133229533-68dfa5df-02b0-48b4-9684-b47a24953c17.png)

I already have `?attr/colorPrimary` and `?attr/colorOnPrimary` and `?android:attr/colorBackground` but in case of top bar I have to use different colors if I'd like to follow Material design

**Comment by svbutko** [comment]
And another thing is that it works on iOS if I'd like to work with built-in colors

**Comment by swabbass** [comment]
On both platforms, you need to choose either the use of JS declared colours or Native declared colours:
if you want to support dark/light colors via native colours, you should declare native colours per platform as the platform suggests to declare them and use `color : PlatformColro(nativeResId)`, and if you want to support light/dark from JS you can use JS values like `color:{ light:'black' , dark:'#F20FFF'}`. this is the suggested way to do that.

TL;DR
`PlatformColor` in ReactNative is parsed natively, hence, they do not return the actual value in JS in order to use it in our solution for dark mode (the JS one), which we did in order to have JS declared colors without going to native, and for those who want to use native scheme for dark mode, we added support for `PlatfromColor` as in native you will pass 1 resource id 
for colour that have dark/light values and when mode changes, the resource value changes by the Theme selection done by the system.

The problem with such usage i can explain with this:
Suppose that there are 2 native resources that supports 2 modes `native1` and `native2`, passing them as:
`color: {light: PlatformColor(native1), dark: PlatformColor(native2)}` natively translate to `{color : light :{ dark:native1, light:native1}, dark:{ dark:native2, light:native2}}` which is wasteful, since only light-light and dark-dark will be used.


now, on Android, if you have `?attr/colorPrimary` and `?attr/colorOnPrimary` you can define values in native XML inside `values` and `values-night`, [see the path of this file for dark ](https://github.com/material-components/material-components-android/blob/master/lib/java/com/google/android/material/theme/res/values-night/themes_daynight.xml) , and [see the same file for no-dark](https://github.com/material-components/material-components-android/blob/master/lib/java/com/google/android/material/theme/res/values/themes_daynight.xml). 

On iOS, `PlatformColor` uses system-defined colours, hence, custom colours won't work, and you can use `DynamicColorIOS` as in [this](https://github.com/wix/react-native-navigation/pull/7248/files#diff-ab91058d1ac2debf264b4c162b20112b7fca5de00a87ea2b37c39a1bcc8303f6R580).
or use it like in this [case](https://github.com/facebook/react-native/blob/main/packages/rn-tester/js/examples/PlatformColor/PlatformColorExample.js#L113), for system defined colors, that has already defined 2 modes for color.
